frontend/catalyst/jax_tracer.py

```diff
@@ -781,35 +785,24 @@ def is_leaf(obj):
                 qreg_in = qalloc_p.bind(len(device.wires))
                 qrp_out = trace_quantum_tape(tape, device, qreg_in, ctx, trace)
                 meas, meas_trees = trace_quantum_measurements(device, qrp_out, output, trees)
-                out_quantum_tracers = [qrp_out.actualize()]
-                qdealloc_p.bind(qrp_out.base)
-
-                tracers = [trace.full_raise(m) for m in meas]
-                results_tracers += tracers
-
-                jaxpr, out_type, _ = ctx.frames[trace].to_jaxpr2(tracers + out_quantum_tracers)
-                jaxpr._outvars = jaxpr._outvars[:-1]  # pylint: disable=protected-access
-                out_type = out_type[:-1]
-
-                out_avals, _ = unzip2(out_type)
-                abstract_results = tree_unflatten(
-                    meas_trees,
-                    [ShapeDtypeStruct(a.shape, a.dtype, a.named_shape) for a in out_avals],
-                )
-                # This mimics the return type from qnodes.
-                # I would prefer if qnodes didn't have special rules about whether they return a
-                # tuple, list, or value.
```

Comment:
Much better! Was this just not needed, or where did this logic moved to?

Reply:
Yes, this is interesting:
1. `to_jaxpr2` is not needed because we can pass the tracers that we have to the post-processing function and generate the final jaxpr there.
2. Trimming the quantum register from jaxpr.outvars: I thought that it was required but I can't remember the exact reason why. Simply omitting the quantum tracer works just as well. The q.register workflow is still produced because we use it in Jaxpr equations.
3. `ShapeDtypeStruct` combines the PyTree-shape and the static-array type of the result. It worked because their sized matched. With the dynamic API we have implicit arguments that do not have positions in PyTree-shape, so this style can not be used. This is why we need to return out_type and out_tree separately.